lib/src/services/api.dart

```diff
@@ -532,8 +534,23 @@ class Api {
       );
     }
   }
-}
 
+  Future<ApiResponse> endUserBalance(String mobileNumber) async {
+    this._autoUpdateReference();
+    http.Response response;
+    try {
+      String url =
+          _ROOT_ENDPOINT + _API_VERSION + _END_USER_BALANCE + mobileNumber;
+      response = await http.get(url, headers: this._headers);
```

Comment:
hot-recharge have a status code of `2` that signals a proper `success` you might want to check that first before assuming that its a good success response like i did in most of my api methods [example](https://github.com/DonnC/hot-recharge-flutter/blob/d9f2ed715eb8f4e80cbe9b28d4710844047771ad/lib/src/services/api.dart#L119) 
You can also add a `_log(...)` method on strategic points in your methods like [here](https://github.com/DonnC/hot-recharge-flutter/blob/d9f2ed715eb8f4e80cbe9b28d4710844047771ad/lib/src/services/api.dart#L117) this way it makes it easy to see logs as you test and also when another developer enables logger via `enableLogger: true` while testing just in case, to give a detailed view.
This is because the hot -recharge api might perform internal changes that the api wont have at the moment so that will help us see what and when to change while testing or someone else might pick it up and notify us.

Reply:
You can do this later on other methods, it looks like the right thing to do for proper response @iamngoni